internal/sdk/cloudian/sdk_test.go

```diff
@@ -1,170 +1,148 @@
+		json.NewEncoder(w).Encode(groupInternal{GroupID: "QA", Active: "true"})
+	})
+	defer testServer.Close()
+
+	expected := Group{
+		GroupID: "QA",
+		Active:  true,
 	}
 
-	if users[0].UserID != "Glory" {
-		t.Errorf("Expected Glory as the userId of first user, got %v", users[0].UserID)
+	group, err := cloudianClient.GetGroup(context.TODO(), "QA")
+
+	if err != nil {
+		t.Errorf("Error getting group: %v", err)
 	}
 
-	if users[1].UserID != "John" {
-		t.Errorf("Expected John as the userId of second user, got %v", users[1].UserID)
+	if diff := cmp.Diff(expected, *group); diff != "" {
+		t.Errorf("GetGroup() mismatch (-want +got):\n%s", diff)
 	}
 
 }
 
-func (group groupInternal) Generate(rand *rand.Rand, size int) reflect.Value {
-	return reflect.ValueOf(groupInternal{
-		Active:             "true",
-		GroupID:            randomString(16),
-		GroupName:          randomString(32),
-		LDAPEnabled:        true,
-		LDAPGroup:          randomString(8),
-		LDAPMatchAttribute: randomString(8),
-		LDAPSearch:         randomString(8),
-		LDAPSearchUserBase: randomString(8),
-		LDAPServerURL:      randomString(8),
-		LDAPUserDNTemplate: randomString(8),
-		S3EndpointsHTTP:    []string{randomString(8), randomString(8)},
-		S3EndpointsHTTPS:   []string{randomString(8), randomString(8)},
-		S3WebSiteEndpoints: []string{randomString(8), randomString(8)},
+func TestGetGroupNotFound(t *testing.T) {
+	cloudianClient, testServer := mockBy(func(w http.ResponseWriter, r *http.Request) {
+		w.WriteHeader(http.StatusNoContent)
 	})
-}
+	defer testServer.Close()
 
-func TestGenericError(t *testing.T) {
-	err := errors.New("Random failure")
+	_, err := cloudianClient.GetGroup(context.TODO(), "QA")
 
-	if errors.Is(err, ErrNotFound) {
-		t.Error("Expected not to be ErrNotFound")
+	if !errors.Is(err, ErrNotFound) {
+		t.Errorf("Expected error to be ErrNotFound")
 	}
 }
 
-func TestWrappedErrNotFound(t *testing.T) {
-	err := fmt.Errorf("wrap it: %w", ErrNotFound)
+func TestCreateCredentials(t *testing.T) {
+	expected := SecurityInfo{AccessKey: "123", SecretKey: "abc"}
+	cloudianClient, testServer := mockBy(func(w http.ResponseWriter, r *http.Request) {
+		json.NewEncoder(w).Encode(expected)
+	})
+	defer testServer.Close()
 
-	if !errors.Is(err, ErrNotFound) {
-		t.Error("Expected to be ErrNotFound")
+	credentials, err := cloudianClient.CreateUserCredentials(context.TODO(), User{GroupID: "QA", UserID: "user1"})
+
+	if err != nil {
+		t.Errorf("Error creating credentials: %v", err)
+	}
+
+	if diff := cmp.Diff(expected, *credentials); diff != "" {
+		t.Errorf("CreateUserCredentials() mismatch (-want +got):\n%s", diff)
 	}
 }
 
-func TestGroupSerialization(t *testing.T) {
-	f := func(group groupInternal) bool {
-		data, err := json.Marshal(group)
-		if err != nil {
-			return false
-		}
+func TestGetUserCredentials(t *testing.T) {
+	expected := &SecurityInfo{AccessKey: "123", SecretKey: "abc"}
```

Comment:
Why pointer in this one and not the previous? What's the difference?

Reply:
no diff - changed to expected as a value in both